arwen/host/asyncCall.go

```diff
@@ -51,21 +49,38 @@ func (host *vmHost) handleAsyncCallBreakpoint() error {
 	return nil
 }
 
-func (host *vmHost) canExecuteSynchronouslyOnDest(dest []byte) bool {
+func (host *vmHost) canExecuteSynchronously(destination []byte, data []byte) bool {
+	runtime := host.Runtime()
 	blockchain := host.Blockchain()
-	calledSCCode, err := blockchain.GetCode(dest)
 
-	return len(calledSCCode) != 0 && err == nil
-}
+	// Sync execution requires the destination to be in the same shard.
+	shardOfSC := blockchain.GetShardOfAddress(runtime.GetSCAddress())
+	shardOfDest := blockchain.GetShardOfAddress(destination)
+	if shardOfSC != shardOfDest {
```

Comment:
in case of built in functions those has to be processed always in both sender and destination shard as well.

Reply:
Indeed. Will change.